internal/lock/lockfile.go

```diff
@@ -46,6 +50,26 @@ func GetFile(project devboxProject) (*File, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	// If the lockfile has legacy StorePath fields, we need to convert them to the new format
+	for pkgName, pkg := range lockFile.Packages {
+		for sys, sysInfo := range pkg.Systems {
+			// If we have a StorePath and no Outputs, we need to convert to the new format.
+			// Note: for a non-empty StorePath, Outputs should be empty, but being cautious.
+			if sysInfo.StorePath != "" && len(sysInfo.Outputs) == 0 {
+				lockFile.Packages[pkgName].Systems[sys].Outputs = []*Output{
+					{
+						Default: true,
+						Name:    "out",
+						Path:    sysInfo.StorePath,
+					},
+				}
+				// Track that we read from the legacy format
+				lockFile.legacySystemsFormat = true
```

Comment:
Can't we use the existence of `StorePath` to tell this? To generalize this boolean (or even get rid of it) we could:

The no boolean solution looks like:

* If lockfile contains store paths it is legacy.
* If lockfile is dirty (i.e. loaded version doesn't match file) we modernize.  (small perf hit for re-reading file, but we should rarely write to lock file and when we do, it's a slower operation anyway)

Version keeping a bool:

* If lockfile contains store paths it is legacy.
* Add `upgradeLockfile` bool that gets triggered manually during add/remove/update.